### DISCUSSION TITLE
How to implement multipart/form-data

### DISCUSSION THREAD

**Discussion by rengas** [go-resty/resty#591] [Q&A]
I have this code but I get  **multipart nextpart eof_ error**   on the server

Will the boundry get added automatically to each part? 
I don't see much document how Multipart works. Any help will apppreciated?

Thanks 

```
func updateImageLocal(token string, restaurantId string, t string, r io.Reader) (interface{}, int) {
	url := fmt.Sprintf("http://localhost:%s/api/v1/%s/image", "8080", id)
	var okRes models.response
	var errRes models.error
	client := resty.New().EnableTrace()

	// POST JSON string
	// No need to set content type, if you have client level setting
	resp, err := client.R().
		SetHeader("Authorization", token).
		SetHeader("Content-Type", "multipart/form-data; boundary=---WebKitFormBoundary7MA4YWxkTrZu0gW").
		SetMultipartFormData(map[string]string{
			"type": t,
		}).
		SetMultipartField("param", "content", "image/png", r).
		SetResult(&okRes).
		SetError(&errRes).
		Patch(url)
	if err != nil {
		logging.Fatal(err.Error())
	}

	if resp.StatusCode() != http.StatusOK {
		return errRes, resp.StatusCode()
	}

	return okRes, resp.StatusCode()
}
```

**Comment by kecci** [comment]
I think there is no problem with your form-data, the error was the `eof`. (_end of file_ or empty data)

Noticing `r io.Reader` as the parameter, is the reader connection you pass was not closed ? is there any defer close before that ?

Try to change your reader with the object or byte, then convert to reader when inside the func of `func updateImageLocal`.

Hope this could be help you
